### ISSUE TITLE
Feature request: conditional nested sub-filters

### ISSUE THREAD

**Issue by rauanmayemir** [spiral/framework#310]
Currently, if a request has a lot of conditional nested fields, mapping and validating them can get very messy or straight up impossible.

The proposed idea is having dependent variants in their own filters and map them to the value 'statically'. Here's an example gist for pseudo schema:

```
public const SCHEMA = [
    'type' => 'data:type',
    'element' => [
        'nested' => 'data:type',
        'mapping' => [
            'type1' => ElementType1Filter::class,
            'type2' => ElementType2Filter::class,
        ]
    ],
];
```

So if the `type` equals `type1`, filter's `element` field will be an instance of `ElementType1Filter` and all the validation related to `type1` will happen inside `ElementType1Filter`.

**Comment by vvval** [comment]
Let me ask If understood it correctly:
```php
public const SCHEMA = [
    'type' => 'data:type',
    'element' => [
        'dependsOn' => 'data:type', 
        'match' => [
            'type1' => ElementType1Filter::class,
            'type2' => ElementType2Filter::class,
        ]
    ],
];
```
the difference is using special keys `dependsOn` and `match` (to look similar to a new match construction). I assume these 2 keywords are more understandable

**Comment by rauanmayemir** [comment]
Yeah, this would work just as fine. Should it depend on ‘data:type’ instead of ‘type’?

**Comment by vvval** [comment]
> Yeah, this would work just as fine. Should it depend on ‘data:type’ instead of ‘type’?

@rauanmayemir if we always refer to a sibling field, then it's ok without `data:`
